content/docs/concepts/low-power.md

```diff
@@ -0,0 +1,64 @@
+The main caveat here is if not configured properly, microcontroller and periferals consume power as usual while your program "sleeps".
+
+To address that, most microcontrollers have one or more "low power" modes, also known as "deep sleep" and, sometimes, "dormant".  
+In such modes microcontroller may consume as low as milli- or even micro-Watts.  
+
+## Implementations
+
+Manufacturers implement different modes, depending of capabilites of their products and name these modes differently too.
+Hence no naming consistency across microcontrollers, unfortunately.
+
+Always consult "Power consumption" part of a respective datasheet to understand capabilities of your hardware.  
+When reading datasheets, keep in mind peripherals that draw power too. These include \[power\] LEDs, various sensors or co-processors usually present on development boards.
+
+Capabilities of different chips:
+- Some very good chips (like the nRF series) can go to a very low power state (just a few µA) with RAM retention and waking up from a low-power RTC.
+- Some limited chips (like the esp32) can't really "sleep" with low power -- they can disable the core and memory and reset after a predefined time.
+- Other chips need to use a special feature to go into deep sleep mode.
+
+Especially the distinction between waking up with or without RAM is important, because it greatly affects how you write software.
+
+The common way to wake up from deep sleep is to use [interrupts](compiler-internals/interrupts), such as timer, real-time-clock or an interrupt on a pin change.
+
+### RP2040
+
+Minimal achievable power consumption according to [RP2040 datasheet](https://datasheets.raspberrypi.com/rp2040/rp2040-datasheet.pdf): 6mW to 12mW (0.19mA at 3.3V)
+
+### nRF Family
+
+[nRF series chips](https://www.nordicsemi.com/Products) are exceptionally efficient and is recommended choice for low-power projects.
+
+Minimal achievable power consumption is in the area of micro-Watts.
+
+Even with BLE broadcasting enabled it can be lower power than rp2040 in its lowest power state!
+
+The things needed to make nrf series chips efficient are:
+- Disable the UART with help of `-serial=none` flag;
+- Use the special sleep function of the SoftDevice;
+
+## The Way to Low Power
+
+Following steps shall help you achieve low power consumption:
+
+- Put a for `{ arm.Asm("wfi") }` loop before anything else is initialized (at the beginning of `Reset_Handler`).
+- Measure the current, this should be the lowest consumption as advertised for the chip.
+- If it is higher than that, investigate why (it could be a power LED, regulator, some clock that's enabled by the bootloader, maybe other things).  
+  Usually this ranges somewhere between 1µA and 100µA, depending on the low-power capabilities of the chip. Don't continue until you really do reach the lowest possible power consumption.
+- Slowly move the loop to later in the code, until you move it into the main function.
```

Comment:
Might be helpful for the benefit of casual hobbyists to discuss (either here or line 59) what the loop is doing, and what "wfi" means (wait for instruction?). 

Also, where to look for the functions that will be run 'before' main(). My best guess is that you're referencing `src/runtime/runtime_[chip].go` and testing the infinite 'wfi' loop before each call in the `init` function. (I will definitely try this!)

Reply:
Yes, you are right, such `init` function you can find by search for "export Reset_Handler" in tinygo repository and picking your architecture. I've added change suggestion on line 59, shall be easier to grasp now.

"wfi" pauses processor in "wait for interrupt", that basically makes processor "do nothing". And if inserted before any initialisation, we can be sure no any internal subsystem is running consuming power.